### PR TITLE
use inspektor gadget

### PR DESCRIPTION
**Pull request by matthyx**

### CODE REVIEW

File: .github/workflows/inspektor-trigger-e2e-tests.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+name: inspektor-e2e-tests
```
Comment (dwertent) [inline .github/workflows/inspektor-trigger-e2e-tests.yaml:1]: We should not push this to main

File: charts/kubescape-cloud-operator/templates/node-agent/clusterrole.yaml [hunk 2]
```diff
@@ -7,6 +7,9 @@ rules:
 - apiGroups: [""]
   resources: ["pods", "nodes"]
   verbs: ["get", "watch", "list"]
+- apiGroups: [""]
```
Comment (dwertent) [inline charts/kubescape-cloud-operator/templates/node-agent/clusterrole.yaml:10]: Question: Does the nodeAgent need this?

File: charts/kubescape-cloud-operator/templates/node-agent/daemonset.yaml [hunk 3]
```diff
@@ -73,39 +86,6 @@ spec:
         - name: proxy-secret
           secret:
             secretName: {{ .Values.global.proxySecretName }}
-        {{- end }}    
-      initContainers:
```
Comment (dwertent) [inline charts/kubescape-cloud-operator/templates/node-agent/daemonset.yaml:77]: YAY!

File: charts/kubescape-cloud-operator/values.yaml [hunk 4]
```diff
@@ -567,9 +567,8 @@ nodeAgent:
   name: "node-agent"
 
   config: 
-    name: node-agent-config-map
-    maxLearningPeriod: 180 # minutes
-    learningPeriod: 2 # minutes
+    maxLearningPeriod: 3h
```
Comment (dwertent) [inline charts/kubescape-cloud-operator/values.yaml:570]: Make sure to update the documentation as well.

File: charts/kubescape-cloud-operator/values.yaml [hunk 5]
```diff
@@ -583,101 +582,51 @@ nodeAgent:
...
       name: node-agent
-      image: 
-        repository: quay.io/kubescape/node-agent
-        tag: v0.1.75
-        pullPolicy: IfNotPresent
-      env:  
-        - name: CONFIG_ENV_VAR
-          value: "/etc/node-agent/configuration/ConfigurationFile.json"
+      image:
+        repository: quay.io/matthiasb_1/node-agent
```
Comment (dwertent) [inline charts/kubescape-cloud-operator/values.yaml:610]: Well... we need to use the image from `quay.io/kubescape`

File: charts/kubescape-cloud-operator/values.yaml [hunk 6]
```diff
@@ -583,101 +582,51 @@ nodeAgent:
...
+        tag: rewrite
+        pullPolicy: Always
```
Comment (dwertent) [inline charts/kubescape-cloud-operator/values.yaml:612]:
```suggestion
        pullPolicy: IfNotPresent
```

File: charts/kubescape-cloud-operator/values.yaml [hunk 7]
```diff
@@ -583,101 +582,51 @@ nodeAgent:
...
             fieldRef:
               fieldPath: spec.nodeName
       volumeMounts:
-        - mountPath: /etc/node-agent/configuration
-          name: configmap-volume
-        - mountPath: /root/.falco
-          name: root-falco-fs
-        - mountPath: /host/proc
-          name: proc-fs
+        - mountPath: /host
```
Comment (dwertent) [inline charts/kubescape-cloud-operator/values.yaml:619]: Do we really need all of this in the values file? Can we have this hard-coded?

**Review by dwertent — CHANGES_REQUESTED** [review]
I think the only blocking comment is the origin of the image. Let's have it hosted in the Kubescape org.

File: .github/workflows/inspektor-trigger-e2e-tests.yaml (hunk 1 above)
Comment (matthyx) [reply .github/workflows/inspektor-trigger-e2e-tests.yaml:1]: @rcohencyberarmor ?

File: charts/kubescape-cloud-operator/templates/node-agent/clusterrole.yaml (hunk 2 above)
Comment (matthyx) [reply charts/kubescape-cloud-operator/templates/node-agent/clusterrole.yaml:10]: yes inspektor gadget needs it

File: charts/kubescape-cloud-operator/values.yaml (hunk 5 above)
Comment (matthyx) [reply charts/kubescape-cloud-operator/values.yaml:610]: yes, once released I'll update the chart

File: charts/kubescape-cloud-operator/values.yaml (hunk 7 above)
Comment (matthyx) [reply charts/kubescape-cloud-operator/values.yaml:619]: it was like that previously, so I just updated the values
we can definitely move that to the daemonset.yaml

File: .github/workflows/inspektor-trigger-e2e-tests.yaml (hunk 1 above)
Comment (dwertent) [reply .github/workflows/inspektor-trigger-e2e-tests.yaml:1]: I added this for the inspektor branch...

File: charts/kubescape-cloud-operator/values.yaml (hunk 5 above)
Comment (dwertent) [reply charts/kubescape-cloud-operator/values.yaml:610]: Dont forget to add the `release` label.
Also, you probably need to update the CI.
Maybe for now, it is better if you re-tag and push

**Review by dwertent — APPROVED** [review]

**Pull request merged**